vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors_moe.py

```diff
@@ -297,15 +299,20 @@ def create_weights(self, layer: torch.nn.Module, num_experts: int,
         layer.register_parameter("w2_weight_packed", w2_weight)
         set_weight_attrs(w2_weight, extra_weight_attrs)
 
-        self.is_k_full = (intermediate_full == intermediate_size)
-        scales_size = (intermediate_full if self.actorder
-                       and self.group_size != -1 else intermediate_size)
+        # In the case where we have actorder/g_idx,
+        # we do not partition the w2 scales
+        load_full_w2 = self.actorder and self.group_size != -1
+        w2_scales_size = (intermediate_full
+                          if load_full_w2 else intermediate_size)
+        # @eliza TODO: is this condition actually needed/is it doing anything?
```

Comment:
This should be used by `fused_marlin_moe()`, but I'd also change it to
```
!self.actorder || intermediate_full == intermediate_size
```

Reply:
sounds good